pkg/commands/workload.go

```diff
@@ -450,8 +449,19 @@ func (opts *WorkloadOptions) PublishLocalSource(ctx context.Context, c *cli.Conf
 		return true, nil
 	}
 
+	var taggedImage string
 	var okToPush bool
-	taggedImage := strings.Split(workload.Spec.Source.Image, "@sha")[0]
+
+	isLocal := opts.SourceImage == ""
```

Comment:
for setting islocal true and update the source image url ,we might have to consider if it is creation or updating as well as the annotation.